internal/services/availability_status_service.go

```diff
@@ -18,12 +18,8 @@ func AvailabilityStatus(w http.ResponseWriter, r *http.Request) {
 	}
 
 	asm := kafka.AvailabilityStatusMessage{SourceID: payload.SourceID}
-	msg, err := asm.GenericMessage(r.Context())
-	if err != nil {
-		renderError(w, r, payloads.NewRenderError(r.Context(), "cannot construct message", err))
-		return
-	}
-	background.EnqueueAvailabilityStatusRequest(&msg)
-	metrics.IncTotalReceivedAvailabilityCheckReqs(err)
+
+	identity := identity.GetIdentityHeader(r.Context())
+	background.EnqueueAvailabilityStatusRequest(identity, &asm)
```

Comment:
It is a minor thing, but I'd say context and extract identity in the `EnqueueAvailabilityStatusRequest` body.
Just a feeling, no blocker :)